syncer/main.py

```diff
@@ -54,137 +127,369 @@ def __init__(self, webinterface, connector):
         self._config = self.read_config(Syncer.default_config)
         self._config_checker = PluginConfigChecker(Syncer.config_description)
 
-        self._token = None
+        self._polling_interval = 60
+        self._gateways = {}
+        self._name = ''
+        self._output_mapping = {}
+        self._shutter_mapping = {}
         self._enabled = False
-        self._previous_outputs = set()
-        self._read_config()
+        thread = Thread(target=self._read_config)
+        thread.start()
+
+        self.connector.output.subscribe_status_event(handler=self.handle_output_status, version=2)
+        self.connector.shutter.subscribe_status_event(handler=self.handle_shutter_status, version=2)
 
         logger.info("Started Syncer plugin")
 
     def _read_config(self):
-        self._ip = self._config.get('gateway_ip', '')
-        self._username = self._config.get('username', '')
-        self._password = self._config.get('password', '')
+        self._enabled = False
+        self._polling_interval = self._config.get('polling_interval', 60)
+        self._name = self._config.get('local_name', '')
+        self._gateways = {}
+        output_mapping = {}
+        shutter_mapping = {}
+        _enabled = False
+        for gateway in self._config.get('gateways', ''):
+
+            ip = gateway.get('gateway_ip', '')
+            username = gateway.get('username', '')
+            password = gateway.get('password', '')
+
+            if gateway.get('remote_name', '') == '':
```

Comment:
gw_name = gateway.get('remote_name', ip)

Reply:
@hendrikVandeput that's not the same. We want to rule out the case where remote name is an empty string. Your statement will only assign the second argument to gw_name whenever if it's None, but this will never be possible as remote_name is a string type.

Reply:
Then I would change the test to make that more explicit, currently it's not clear.